fastplotlib/graphics/_axes.py

```diff
@@ -0,0 +1,423 @@
+        return self._follow
+
+    @follow.setter
+    def follow(self, value: bool):
+        if not isinstance(value, bool):
+            raise TypeError
+        self._follow = value
+
+    def update_bounded(self, bbox):
+        """update the axes using the given bbox"""
+        self._update(bbox, (0, 0, 0))
+
+    def auto_update(self):
+        """
+        Auto update the axes
+
+        For orthographic projections of the xy plane, it will calculate the inverse projection
+        of the screen space onto world space to determine the current range of the world space
+        to set the rulers and ticks
+
+        For perspective projetions it will just use the bbox of the scene to set the rulers
+
+        """
+
+        rect = self._plot_area.get_rect()
+
+        if self._plot_area.camera.fov == 0:
+            # orthographic projection, get ranges using inverse
+
+            # get range of screen space
+            xmin, xmax = rect[0], rect[2]
+            ymin, ymax = rect[3], rect[1]
+
+            world_xmin, world_ymin, _ = self._plot_area.map_screen_to_world((xmin, ymin))
+            world_xmax, world_ymax, _ = self._plot_area.map_screen_to_world((xmax, ymax))
+
+            world_zmin, world_zmax = 0, 0
+
+            bbox = np.array([
+                [world_xmin, world_ymin, world_zmin],
+                [world_xmax, world_ymax, world_zmax]
+            ])
+        else:
+            # set ruler start and end positions based on scene bbox
+            bbox = self._plot_area._fpl_graphics_scene.get_world_bounding_box()
+
+        if self.follow and self._plot_area.camera.fov == 0:
+            # place the ruler close to the left and bottom edges of the viewport
+            # TODO: determine this for perspective projections
+            xscreen_10, yscreen_10 = 0.1 * rect[2], 0.9 * rect[3]
+            edge_positions = self._plot_area.map_screen_to_world((xscreen_10, yscreen_10))
+
+        else:
+            # axes intersect at the origin
+            edge_positions = 0, 0, 0
+
+        self._update(bbox, edge_positions)
+
+    def _update(self, bbox, ruler_intersection_point):
+        """update the axes using the given bbox and ruler intersection point"""
+
+        world_xmin, world_ymin, world_zmin = bbox[0]
+        world_xmax, world_ymax, world_zmax = bbox[1]
+        world_x_10, world_y_10, world_z_10 = ruler_intersection_point
+
+        # swap min and max for each dimension if necessary
+        if self._plot_area.camera.local.scale_y < 0:
+            world_ymin, world_ymax = world_ymax, world_ymin
+            self.y.tick_side = "right"  # swap tick side
+            self.x.tick_side = "left"
+        else:
+            self.y.tick_side = "left"
+            self.x.tick_side = "right"
+
+        if self._plot_area.camera.local.scale_x < 0:
+            world_xmin, world_xmax = world_xmax, world_xmin
+            self.x.tick_side = "left"
```

Comment:
@almarklein is there a nicer way to do this? :sweat_smile:

Reply:
Can probable be made shorter, but not sure if it'd be prettier :)

Does this also do the right thing if `scale_x` and `scale_y` are negative?

Reply:
we'll figure this out later